server/pom.xml

```diff
@@ -11,6 +11,67 @@
   <artifactId>keywhiz-server</artifactId>
   <name>Keywhiz Server</name>
 
+  <profiles>
+    <profile>
+      <id>container</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>sql-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>create-keywhiz-database</id>
+                <phase>none</phase>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
```

Comment:
It turns out that about half of the Keywhiz tests are actually integration tests that rely on being able to connect to a local MySQL database. It also turns out that the server sub-project implicitly relies on the model sub-project (which builds first because the server sub-project depends on it) to create the test database rather than handling this directly. Now that that's no longer the case, the database setup "code" has been copied over to this POM.

That's all fine and dandy, but there are some cases (containerized builds, for example, thus the profile name) where we don't want to have a database dependency at all. Although we can attain this for the most part by bypassing tests via passing `-Dmaven.test.skip=true` to Maven on the command line, it turns out this property is really more of a by-convention thing rather than something first-class that actually de-activates all testing-related lifecycle phases. In other words, it doesn't affect our plugin in any way, which means the plugin will run and fail the build if the database is unreachable. And so, in order to gives ourselves an out, we can use a profile to manually disable the plugin (phase `none`) and then select that profile from the command line via `-P container`.

Reply:
maybe add a small comment about "phase `none`" that it disables the plugin?

Reply:
Done!